src/framework/uicomponents/qml/MuseScore/UiComponents/TimeInputField.qml

```diff
@@ -0,0 +1,81 @@
+import QtQuick 2.15
+
+import MuseScore.Ui 1.0
+
+Row {
+    id: root
+
+    property date time // format: h:mm:ss:ms
+
+    spacing: 4
+
+    opacity: enabled ? 1 : ui.theme.itemOpacityDisabled
+
+    NumberInputField {
+        id: hoursField
+
+        maxValue: 9
+
+        value: root.time.getHours()
+
+        onValueChanged: {
+            var newTime = root.time
+            newTime.setHours(value)
+            root.time = newTime
+        }
+    }
+
+    StyledTextLabel {
+        anchors.verticalCenter: parent.verticalCenter
+        text: ":"
+        font: hoursField.font
+    }
+
+    NumberInputField {
+        maxValue: 60
+
+        value: root.time.getMinutes()
+
+        onValueChanged: {
+            var newTime = root.time
+            newTime.setMinutes(value)
+            root.time = newTime
+        }
+    }
+
+    StyledTextLabel {
+        anchors.verticalCenter: parent.verticalCenter
+        text: ":"
+        font: hoursField.font
+    }
+
+    NumberInputField {
+        maxValue: 60
+
+        value: root.time.getSeconds()
+
+        onValueChanged: {
+            var newTime = root.time
+            newTime.setSeconds(value)
+            root.time = newTime
+        }
+    }
+
+    StyledTextLabel {
+        anchors.verticalCenter: parent.verticalCenter
+        text: ":"
+        font: hoursField.font
+    }
+
+    NumberInputField {
```

Comment:
I believe we should also pay attention to validation here

like seconds value shouldn't be bigger than 60 and I don't see any validation for now, is it just me?

Reply:
That's why I suggest to create a new C++ validator for time  (see uicomponents/view/validators) - so you don't need this new components OR reuse IntInputValidator

Reply:
There is a validation for the maximum value (maxValue: 60)